addons/web/static/src/core/datetime/datetimepicker_service.js

```diff
@@ -467,6 +467,17 @@ export const datetimePickerService = {
                                 editableInputs++;
                             }
                         }
+                        const calendarIconGroupEl = getInput(0)?.parentElement
+                            .querySelector(".input-group-text .fa-calendar")?.parentElement;
```

Comment:
@rdeodoo I'd say there are two remaining things I would improve here:

1. In edit mode, you can change that fa-calendar by another icon (and there are many suitable calendar icons we suggest) and you can also replace it by an image. Maybe we can improve with "search for the input group text with a calendar fa icon, otherwise take the first input-group text" ?

2. In edit mode, clicking that icon should not open the datetimepicker.

Reply:
About 1. I am not sure, I initially did not target the `.fa-calendar` but then noticed snippet like that:
![image](https://github.com/odoo/odoo/assets/30048408/8f3542ab-2cd0-415a-a6d9-c2e7a0223422)
So I guess something like that would make sense too:
![image](https://github.com/odoo/odoo/assets/30048408/d830c7a7-f149-4f11-bd5c-ed975187d568)
And in that case we wouldn't want the search click to open the date picker.

Following 2. and considering one, we could just make it not clickable and not editable at all in edit mode?

In the meantime I pushed some css rule to disable the picker opening and edition of that element

Reply:
> And in that case we wouldn't want the search click to open the date picker.

In that case, it really should be a button and not a input-group-text IMO :wink: 

> In the meantime I pushed some css rule to disable the picker opening and edition of that element

Mmh, why not I guess. Although when we implement our task that allows to upload our own icons, it will be an issue I guess :grimacing: But we'll see at that point I suppose. I guess that, worst case, we can explicitely mark with a class the elements that should open the datepicker in master.

Reply:
Yep, that's what I thought but hardly viable in stable as I would have to patch a few places (cf commit msg) and it would need module update. And we can't add it in js either as... we need to find it first.
But it's definitely something to think about in master:

Note that none of those are editable as of today anyway, so it would need a bit of change regardless of this task:
- Survey is not editable through the builder
- The ecommerce product page rental date picker icon can't be changed
- The rental snippet icon can't be edited either
- The website form date field icon is also not editable

Most of those `<i/>` receive the `contenteditable=false` attribute

Reply:
> That one is definitely editable, that's the one I tested before adding my message here.

It's not? Double clicking on the icon does nothing, and clicking once does not show any option to do so, neither on Chrome and neither on Firefox:
![image](https://github.com/odoo/odoo/assets/30048408/c9eb7ea6-edff-4348-8f89-de4d2af3c361)

Reply:
Ok it was a nice side-effect of your original diff when I tested it: 
![image](https://github.com/odoo/odoo/assets/10338094/0b2f3606-1be4-48c8-a281-5e2c9210a80b)

In any case, it's not forced: it could be editable anywhere on the website. It is not in the snippet because we mark it entirely non editable except submit button and descriptions.

Anyways, let's move on, I'll retest and merge